### PR TITLE
fix(core): preserve package.json indent

### PR DESCRIPTION
**Pull request by lihbr**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Currently `siroc` was overriding indentation of the `package.json` file when writing it back to file system, having it forced to two spaces. This is not convenient for users with a preference for different indentation strategies (4 spaces, tabs, etc.).

After speaking with @danielroe we agreed on using [`detect-indent`](https://www.npmjs.com/package/detect-indent) to preserve current indentation when writing back the definition file.

I tested on a project of mine using spaces, `siroc` now behaves as expected ☺️

As always let me know if anything! Thanks for `siroc`! Cheers! win92

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by lihbr** [comment]
@danielroe? 🥺

**Comment by danielroe** [comment]
@lihbr Mea culpa! 👀

File: src/core/package/index.ts [hunk 1]
```diff
@@ -215,8 +217,9 @@ export class Package {
    */
   async writePackage() {
     const pkgPath = this.resolvePath('package.json')
+    const indent = detectIndent(readFileSync(pkgPath, 'utf8')).indent || '  '
```
Comment (danielroe) [inline src/core/package/index.ts:220]: Ideally I'd like to detect this when the package is initialised as that's when we read in the package.json.

Comment (lihbr) [reply src/core/package/index.ts:220]: Ok, and then store it on a property to be used there again? Happy to update

Comment (danielroe) [reply src/core/package/index.ts:220]: Exactly 👍

Comment (lihbr) [reply src/core/package/index.ts]: efe77dfe771c9d3d819a3c0ca4eed7ce5628b50a should be good :)

I doubled check what we might be missing out on not using [readJSONSync](https://github.com/jprichardson/node-jsonfile/blob/master/index.js#L40-L61), we're not using the "fancy" options, and we're enforcing `utf8` so from my understanding `stripBom` doesn't help us with anything(?)

Let me know :)

**Review by danielroe — APPROVED** [review]
Thanks!

**Pull request merged**